user/User.cc

```diff
@@ -1,135 +1,83 @@
 #include "User.h"
+#include "../utils/StatusPrinter.h"
+#include "../devices/AirConditioner.h"
+#include "../devices/Heater.h"
+#include "../devices/Light.h"
 
 namespace SmartHome {
-    User::User(int numRooms, int numSensorsPerRoom) : userHouse(numRooms, numSensorsPerRoom) {
-        devices.push_back(new Device(LIGHT));
-        devices.push_back(new Device(HEATER));
-    }
-
-    User::User(const User& other) : userHouse(other.userHouse) {
-        devices.reserve(other.devices.size());
-        for (Device* device : other.devices) {
-            devices.push_back(new Device(*device));
-        }
-    }
-
-    User::User(User&& other) : userHouse(std::move(other.userHouse)), devices(std::move(other.devices)) {
-        other.devices.clear();
-    }
-
-    User& User::operator=(User&& other) {
-        if (this != &other) {
-            userHouse = std::move(other.userHouse);
-
-            for (Device* device : devices) {
-                delete device;
-            }
-            devices = std::move(other.devices);
-            other.devices.clear();
-        }
-        return *this;
-    }
-
-    User& User::operator=(const User& other) {
-        if (this != &other) {
-            userHouse = other.userHouse;
-
-            for (Device* device : devices) {
-                delete device;
-            }
-            devices.clear();
-
-            devices.reserve(other.devices.size());
-            for (Device* device : other.devices) {
-                devices.push_back(new Device(*device));
-            }
-        }
-        return *this;
-    }
-
-    User::~User() {
-        for (Device *device: devices) {
-            delete device;
-        }
+    User::User(int numRooms) : userHouse(numRooms) {
+        devices.push_back(new Light());
+        devices.push_back(new Heater());
+        devices.push_back(new AirConditioner());

```

Comment:
you should have left the previous implementation here as now you have memory leaks due to not deleting the devices.

Reply:
Will add a destructor to the User class